pkg/labels/selector.go

```diff
@@ -809,3 +809,73 @@ func SelectorFromSet(ls Set) Selector {
 func ParseToRequirements(selector string) ([]Requirement, error) {
 	return parse(selector)
 }
+
+// Conflicts takes 2 maps
+// returns true if there a key match between the maps but the value doesn't match
+// returns false in other cases
+func Conflicts(labels1, labels2 map[string]string) bool {
+	for k, v := range labels1 {
+		if val, match := labels2[k]; match {
+			if val != v {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// Merge combines given maps
+// Note: It doesn't not check for any conflicts between the maps
+func Merge(labels1, labels2 map[string]string) map[string]string {
+	mergedMap := map[string]string{}
+
+	for k, v := range labels1 {
+		mergedMap[k] = v
+	}
+	for k, v := range labels2 {
+		mergedMap[k] = v
+	}
+	return mergedMap
+}
+
+// Equals returns true if the given maps are equal
+func Equals(labels1, labels2 map[string]string) bool {
+	if len(labels1) != len(labels2) {
+		return false
+	}
+
+	for k, v := range labels1 {
+		value, ok := labels2[k]
+		if !ok {
+			return false
+		}
+		if value != v {
+			return false
+		}
+	}
+	return true
+}
+
+// Converts selector string to labels map and validates keys and values
```

Comment:
Nit: Doc should be in the form of "ConvertSelectorToLabelsMap frobs the bits..." (start with the method name).